tycho-core/src/main/java/org/eclipse/tycho/core/maven/MavenDependencyInjector.java

```diff
@@ -63,9 +68,13 @@ public static void injectMavenDependencies(MavenProject project, DependencyArtif
 
     private final MavenProject project;
 
-    MavenDependencyInjector(MavenProject project, BundleReader bundleReader, Logger logger) {
+    private Function<ArtifactDescriptor, MavenDependencyDescriptor> descriptorMapping;
+
+    MavenDependencyInjector(MavenProject project, BundleReader bundleReader,
```

Comment:
If we can have the method directly in ArtifactDescriptor, then no need to forward a mapper and we get stronger typing.

Reply:
See comment above, this is actually not possible here due to maven<->OSGi bordering ....